app/src/main/java/mozilla/lockbox/store/AccountStore.kt

```diff
@@ -237,8 +240,12 @@ open class AccountStore(
 
         webView.clearCache(true)
         clearLogs()
+    }
 
-        dispatcher.dispatch(DataStoreAction.Reset)
+    private fun removeDeviceFromFxA() {
+        runBlocking {
```

Comment:
Not sure if `run {}` or `runBlocking {}` would be a suitable coroutine to use here. Maybe you have thoughts @sashei

Reply:
whoa, we should never block the thread in regular code! let's stick with `run` for now... or even better, create an Observable to wrap this code so that we can interact with it in the usual way :)

Reply:
Let's pair on that. I'm not sure how to create an Observable for a `Deferred` type.